open-apparel-registry/open-apparel-registry
# Have admins confirm/reject potential matches

## Overview

Moves confirm/rejection of potential facility matches for most lists from being done by contributors to being done by OS Hub admins. Select contributors and/or lists can be changed in the Django admin to be moderated by contributors instead, as will any existing lists.

Also updates the admin list dashboard to add a new option to display all contributors (shown by default), as well as a new control for filtering by who is responsible for match moderation and pagination controls.

Connects #1979

## Demo

![image](https://user-images.githubusercontent.com/4432106/183954172-515bc2b0-c9af-45fc-ac05-97e45b6bfb41.png)


![image](https://user-images.githubusercontent.com/4432106/183954045-c66b254f-1aa7-486f-a3c2-fca59b893855.png)

![image](https://user-images.githubusercontent.com/4432106/183956111-3ce79567-a9d1-42ba-91e5-e82e34b21b08.png)


## Notes

This adds a new default `status` filter of `MATCHED` to the Dashboard - we'll restore the ability to see lists in other status' in https://github.com/open-apparel-registry/open-apparel-registry/issues/2017

## Testing Instructions

* On `develop`
* As a regular contributor, add a new list that will cause some items to be potential matches that need to be confirmed / rejected (I uploaded the same list twice w/ different names)
* On this branch run `scripts/manage migrate`
* You should be able to continue to moderate the lists you already uploaded
* If you upload another list, you should instead see a message that it is awaiting moderation
* Log out and log in as a super user (or log in using a private window). On the Dashboard lists page you should see the list you just uploaded, and be able to confirm/reject the potential match.
 * If you switch the match responsibility filter you should see the first w/ potential matches list you uploaded at the top of the list

## Checklist
